ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -211,24 +227,159 @@ const DataProcConfigSelector = ({onChange, dataprocConfig})  => {
+      </div>
+    }
+    {
+      workspace.billingAccountType === BillingAccountType.FREETIER
+      && profile.username !== workspace.creator
+      && <div style={{borderLeft: `1px solid ${colorWithWhiteness(colors.dark, .5)}`, padding: '.33rem .5rem'}}>
+        Costs will draw from workspace creator's remaining {formatUsd(freeCreditsRemaining)} of free credits.
+      </div>
+    }
+    {
+      workspace.billingAccountType === BillingAccountType.USERPROVIDED
+      && <div style={{borderLeft: `1px solid ${colorWithWhiteness(colors.dark, .5)}`, padding: '.33rem .5rem'}}>
+        Costs will be charged to billing account {workspace.billingAccountName}.
+      </div>
+    }
+  </FlexRow>;
+};
+
+export const RuntimePanel = fp.flow(
+  withCdrVersions(),
+  withCurrentWorkspace(),
+  withUserProfile()
+)(({cdrVersionListResponse, workspace, profileState}) => {
+  const {namespace, id, cdrVersionId} = workspace;
+
+  const {profile} = profileState;
+
   const {hasMicroarrayData} = fp.find({cdrVersionId}, cdrVersionListResponse.items) || {hasMicroarrayData: false};
   const [currentRuntime, setRequestedRuntime] = useCustomRuntime(namespace);
+
   const {status = null, dataprocConfig = null, gceConfig = {diskSize: defaultDiskSize}} = currentRuntime || {} as Partial<Runtime>;
-  const machineName = !!dataprocConfig ? dataprocConfig.masterMachineType : gceConfig.machineType;
-  const diskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.diskSize;
-  const initialMasterMachine = findMachineByName(machineName);
+  const diskSize = dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.diskSize;
+  const machineName = dataprocConfig ? dataprocConfig.masterMachineType : gceConfig.machineType;
+  const initialMasterMachine = findMachineByName(machineName) || defaultMachineType;
+  const initialCompute = dataprocConfig ? ComputeType.Dataproc : ComputeType.Standard;
+
   const [selectedDiskSize, setSelectedDiskSize] = useState(diskSize);
   const [selectedMachine, setSelectedMachine] = useState(initialMasterMachine);
-  const [selectedCompute, setSelectedCompute] = useState<ComputeType>(dataprocConfig ? ComputeType.Dataproc : ComputeType.Standard);
+  const [selectedCompute, setSelectedCompute] = useState<ComputeType>(initialCompute);
   const [selectedDataprocConfig, setSelectedDataprocConfig] = useState<DataprocConfig | null>(dataprocConfig);
 
   const selectedMachineType = selectedMachine && selectedMachine.name;
   const runtimeExists = status && status !== RuntimeStatus.Deleted;
   const runtimeChanged = !fp.equals(selectedMachine, initialMasterMachine) ||
     selectedDiskSize !== diskSize ||
-    !fp.equals(selectedDataprocConfig, dataprocConfig);
+    !fp.equals(selectedDataprocConfig, dataprocConfig) ||
+    !fp.equals(selectedCompute, initialCompute);
+
+  const [creatorFreeCreditsRemaining, setCreatorFreeCreditsRemaining] = useState(0);
+  useEffect(() => {
+    const fetchFreeCredits = async() => {
```

Comment:
nit: correct pattern here is probably to create a `new AbortController()`, pass `{signal: aborter.signal}` on the API fetch, then return `() => aborter.abort()` from this useEffect callback as a cleanup.

This way, if the component gets dismounted, we cancel the fetch.